Mathlib/Algebra/Jordan/Basic.lean

```diff
@@ -122,6 +122,10 @@ instance (priority := 100) CommSemigroup.isCommJordan [CommSemigroup A] : IsComm
   lmul_comm_rmul_rmul _ _ := mul_assoc _ _ _
 #align comm_semigroup.is_comm_jordan CommSemigroup.isCommJordan
 
+-- see Note [lower instance priority]
+instance (priority := 100) IsCommJordan.isCommMul [Mul A] [IsCommJordan A] : CommMul A where
+  mul_comm := IsCommJordan.mul_comm
```

Comment:
@eric-wieser would it be more idiomatic to change the definition of `IsCommJordan` to:
```
class IsCommJordan [CommMul A] : Prop where
  lmul_comm_rmul_rmul : ∀ a b : A, a * b * (a * a) = a * (b * (a * a))
```
?

One would then need to introduce
```
class NonUnitalNonAssocCommRing (α : Type u) extends NonUnitalNonAssocRing α, CommMul α
```
in order to be able to write:
```
variable {A} [NonUnitalNonAssocCommRing A] [IsCommJordan A]
```

Reply:
Yes, I think that would be more idiomatic. The approach here is a bit dangerous because it forms a loop; though perhaps you already got rid of it.

Reply:
@eric-wieser It seems to build okay. Where is the loop?